### PR TITLE
Core/Unit: Fix display for Shaman totems

### PR DESCRIPTION
**Pull request by DoctorKraft**
**Changes proposed:**

What the title say !
- Add display for Pandaren and Highmountain Tauren 
Currently all totems display are broken, probably not updated since lich king

**Target branch(es):** 3.3.5/master

- [X] master

**Tests performed:** (Does it build, tested in-game, etc.)

Tested IG & build

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.h
Comment (Ascathor) [inline src/server/game/Entities/Unit/Unit.h]: Just a small typo: Voodoo Totem

File: src/server/game/Entities/Unit/Unit.h
Comment (Ascathor) [inline src/server/game/Entities/Unit/Unit.h]: Just a small typo: indent is not correct

**Comment by DoctorKraft** [comment]
Fixed travis and reviews

**Comment by tkrokli** [comment]
Looks straight-forward to me. Just a practical tip for Pull Request management:
Do not merge commits after review comments are added unless you do not want approvals.
Merging commits will remove the option to add reviewer approvals to the changes.

**Review by EgoSumThan — APPROVED** [review]

**Comment by tkrokli** [comment]
LoL, oh well. I guess I have missed out on some kind of git functionality. Anyway, I am tired of seeing "We went looking everywhere, but could not find the commits" when the update is a merge of all the previous commits. I guess I'll just unsubscribe then.

File: src/server/game/Entities/Unit/Unit.cpp
Comment (joschiwald) [inline src/server/game/Entities/Unit/Unit.cpp]: that would mean, all other totemcategories are firetotems

Comment (Asaelle) [reply src/server/game/Entities/Unit/Unit.cpp]: Fixed with last commit, i don't have see but TotemCategory are already named in SharedDefine. I know this PR are not 100% blizz because i have two options, create hotfix in SpellTotems.db2 for add category of totem for spell player and this method. Also normaly displayid are done by a serverside spell. So much data in hotfix table...

Comment (DoctorKraft) [reply src/server/game/Entities/Unit/Unit.cpp]: Fixed with last commit, i don't have see but TotemCategory are already named in SharedDefine. I know this PR are not 100% blizz because i have two options, create hotfix in SpellTotems.db2 for add category of totem for spell player and this method. Also normaly displayid are done by a serverside spell. So much data in hotfix table...

File: src/server/game/Entities/Unit/Unit.h
Comment (joschiwald) [inline src/server/game/Entities/Unit/Unit.h]: TOTEM_CATEGORY is a bit misleading, it is a SummonProperty

File: src/server/game/Entities/Unit/Unit.h
Comment (joschiwald) [inline src/server/game/Entities/Unit/Unit.h]: are these old values not longer used? they are still in db2

Comment (DoctorKraft) [reply src/server/game/Entities/Unit/Unit.h]: Only SUMMON_TYPE_TOTEM_WATER are used now for player, after others spells are for creature etc.. so don't need for get the model because it's another stuff

File: src/server/game/Entities/Unit/Unit.h
Comment (DoctorKraft) [reply src/server/game/Entities/Unit/Unit.h]: you prefer with enum SummonProperties and SUMMON_PROPERTY_TOTEM_WATER for example?

Comment (joschiwald) [reply src/server/game/Entities/Unit/Unit.h]: yes

**Comment by Shauren** [comment]
You can't determine this basing on SummonProperties id, http://www.wowhead.com/spell=210643/totem-mastery summons 4 totems (one of each element) and all of them have the same SummonProperties

**Comment by DoctorKraft** [comment]
@Shauren Yes i know.. The problem it's because this inplentation are not 100% blizzlike. All players totem don't have SpellID in SpellTotems.db2 for define category of totem and also the displayid are done by a serverside aura in sniff. Or probably aura are set manually for each totem. I tried to get a method without hotfix data

**Comment by Shauren** [comment]
You are mistaken about SpellTotems.db2 purpose - this db2 defines which item must be present in inventory to cast the spell (wotlk totems, blacksmith hammer etc)

**Comment by DoctorKraft** [comment]
Oh ok, so in retail serverside aura define displayid. I don't have idea for get category of totem in a generic way

**Pull request closed without merging**
